config/management/operational/src/validator_config.rs

```diff
@@ -266,17 +261,18 @@ impl DecryptedValidatorConfig {
         account_address: AccountAddress,
         encryptor: &Encryptor,
     ) -> Result<Self, Error> {
-        let full_node_network_address = NetworkAddress::try_from(&config.full_node_network_address)
+        let full_node_network_addresses = config
+            .full_node_network_addresses(None)
             .map_err(|e| Error::NetworkAddressDecodeError(e.to_string()))?;
 
-        let validator_network_address = encryptor
-            .decrypt(config.validator_network_address.as_ref(), account_address)
+        let validator_network_addresses = encryptor
+            .decrypt(&config.validator_network_addresses, account_address)
             .map_err(|e| Error::NetworkAddressDecodeError(e.to_string()))?;
 
         Ok(DecryptedValidatorConfig {
             consensus_public_key: config.consensus_public_key.clone(),
-            full_node_network_address,
-            validator_network_address,
+            full_node_network_address: full_node_network_addresses[0].clone(),
```

Comment:
why not the vector?

Reply:
I should probably litter the tooling with note or todos, but it doesn't support multiple addresses and is definitely outside of scope for this version.